### ISSUE TITLE
Error when createCohortTables and generateCohortSet on Microsoft SQL server

### ISSUE THREAD

**Issue by cbarbozaerasmus** [OHDSI/DatabaseConnector#215]

**Comment by schuemie** [comment]
As you probably know, Azure doesn't allow you to switch between databases (just schemas). From what I can gather, this error is most likely due to a mismatch in the name of the database on the server, and the database you specified (in `cohortDatabaseSchema`). Importantly, this is a rare situation where the database name is **case sensitive**.

@cbarbozaerasmus : can you check that your database name in `cohortDatabaseSchema` matches exactly the name on the server (including case)?

(PS the error message refers to `USE`, but there is actually no `USE` command being issued).

**Comment by ginberg** [comment]
hi @schuemie, I am also running into the same error as @cbarbozaerasmus on the synthea sql server.  I am wondering if this could be solved by changing my db variables? For synthea there is a schema with your username for write access, so I am using that one for my cohort results. I can see in Azure data studio that these cohort tables have been created in schema "ginberg" (so createCohortTables has been done successfully). This is the code I am using:

```
library(CohortDiagnostics)
library(CohortGenerator)
library(DatabaseConnector)

# Variables
cdmDatabaseSchema <- "cdm_synthea_1M"
cohortDatabaseSchema <- "ginberg"
cohortTable <- "BenchmarkSynthea50"

connectionDetails <- DatabaseConnector::createConnectionDetails(dbms = "sql server",
                                                                user = "ginberg",
                                                                password = Sys.getenv("TESTDB_PWD"),
                                                                server = paste("sqlsrvr-cdm-dev-001.database.windows.net", "database=sql-synthea-1M;encrypt=true;trustServerCertificate=false;hostNameInCertificate=*.database.windows.net;loginTimeout=30;", sep = ";"),
                                                                pathToDriver = Sys.getenv("DATABASECONNECTOR_JAR_FOLDER"))

cohortDefinitionSet <- CohortGenerator::getCohortDefinitionSet(settingsFileName = "Cohorts.csv",
                                                               jsonFolder = "cohorts",
                                                               sqlFolder = "sql/sql_server",
                                                               packageName = "CohortDiagnostics")

cohortTableNames <- CohortGenerator::getCohortTableNames(cohortTable = cohortTable)
CohortGenerator::createCohortTables(connectionDetails = connectionDetails,
                                    cohortTableNames = cohortTableNames,
                                    cohortDatabaseSchema = cohortDatabaseSchema,
                                    incremental = FALSE)
CohortGenerator::generateCohortSet(connectionDetails = connectionDetails,
                                   cdmDatabaseSchema = cdmDatabaseSchema,
                                   cohortDatabaseSchema = cohortDatabaseSchema,
                                   cohortTableNames = cohortTableNames,
                                   cohortDefinitionSet = cohortDefinitionSet,
                                   incremental = FALSE) 
```

**Comment by schuemie** [comment]
@ginberg: can you check the CDM database is indeed exactly called 'cdm_synthea_1M' on your server? (Paying special attention to the case of each letter)

Also, on SQL Server, shouldn't you always have a database and a schema? The default schema is usually 'dbo', so 
```r
cdmDatabaseSchema <- "cdm_synthea_1M.dbo"
```

**Comment by ginberg** [comment]
Actually, the database name is "sql-synthea-1M". The cdmDatabaseSchema is indeed "cdm_synthea_1M". See attached a screenshot from my azure data studio. I don't see a dbo schema, i think Mees might have changed the defaults for this database.

<img width=50% height=50% alt="db_tables" src="https://user-images.githubusercontent.com/7089667/212934181-0597b66e-3e3c-42f2-97f2-320a95fe3082.png">

**Comment by schuemie** [comment]
Ah, in that case I think 
```r
cdmDatabaseSchema <- "sql-synthea-1M.cdm_synthea_1M"
```
But I'm not sure (I don't have an Azure testing database)

**Comment by ginberg** [comment]
hi @schuemie, this doesn't work. I am getting the error
`com.microsoft.sqlserver.jdbc.SQLServerException: Incorrect syntax near '-'.`

For other packages (e.g. FeatureExtraction), I have used "cdm_synthea_1M" successfully as schema name, so without the database name. Also the createCohortTables did succeed, this function just uses my username as schema name. I think because I have specified the server name already in the 'server' argument of createConnectionDetails it's not needed in the schema anymore.
I have asked Mees will create an account for you on the synthea (sql server) test database. Maybe you can have a look at my script using your account?

**Comment by mvankessel-EMC** [comment]
> ```r
> sql-synthea-1M
> ```

Hi Ger, could you try escaping the database name?

Like so:
[sql-synthea-1M].cdm_synthea_1M

A query like this should work as well:
`SELECT TOP 10 person_id FROM [sql-synthea-1M].cdm_synthea_1M.person;`

(From the following [StackOverflow issue](https://stackoverflow.com/questions/20995394/can-i-give-sql-server-database-name-with-hyphen-like-abc-123))

**Comment by schuemie** [comment]
Thanks @ginberg for setting me up with an Azure test account! 

As @mvankessel-EMC points out, it seems to work when you set
```r
cdmDatabaseSchema <- "[sql-synthea-1M].cdm_synthea_1M"
```

**Comment by ginberg** [comment]
yes that works indeed. Thanks for the help @mvankessel-EMC and @schuemie!
